### PR TITLE
Add support for EU TCP endpoint testing

### PR DESCRIPTION
**Pull request by nmuesch**
Rely on the DD_SITE parameter for determining the default of where to test the Logs API. 
Make the default be US if thats not specified.

Use the same approach to know where to send the event.

### CODE REVIEW

File: lib/test-endpoint.sh [hunk 1]
```diff
@@ -2,13 +2,26 @@
 
 unset DD_LOGS_VALID_ENDPOINT
 DATADOG_DIR="${DATADOG_DIR:-/home/vcap/app/datadog}"
+DD_US_API_SITE="https://api.datadoghq.eu/api/"
+DD_EU_API_SITE="https://api.datadoghq.us/api/"
+DD_API_SITE=${DD_US_API_SITE}
+DD_USE_EU=false
```
Comment (gzussa) [inline lib/test-endpoint.sh]: Could we do without this? It seems redundant a little.

File: lib/test-endpoint.sh [hunk 2]
```diff
@@ -2,13 +2,26 @@
 
 unset DD_LOGS_VALID_ENDPOINT
 DATADOG_DIR="${DATADOG_DIR:-/home/vcap/app/datadog}"
+DD_US_API_SITE="https://api.datadoghq.eu/api/"
+DD_EU_API_SITE="https://api.datadoghq.us/api/"
```
Comment (zippolyte) [inline lib/test-endpoint.sh]: it's the other way around :)

File: lib/test-endpoint.sh [hunk 3]
```diff
@@ -2,13 +2,26 @@
...
 
 if [ -z $DD_LOGS_CONFIG_LOGS_DD_URL ]; then
   # Initialize to default value
+  # If DD_SITE contains ".eu" use the EU endpoint, otherwise default to US
   if [ -n $DD_LOGS_CONFIG_DD_PORT -a -n $DD_LOGS_CONFIG_DD_URL ]; then
     DD_LOGS_CONFIG_LOGS_DD_URL="$DD_LOGS_CONFIG_DD_URL:$DD_LOGS_CONFIG_DD_PORT"
   else
-    DD_LOGS_CONFIG_LOGS_DD_URL="agent-intake.logs.datadoghq.com:10516"
+    if [[ $DD_USE_EU ]]; then
+      DD_LOGS_CONFIG_LOGS_DD_URL="agent-intake.logs.datadoghq.eu:443"
```
Comment (zippolyte) [inline lib/test-endpoint.sh]: Is it 443 in EU ? Seems to be the http endpoint no ?

File: lib/test-endpoint.sh (hunk 2 above)
Comment (dabcoder) [reply lib/test-endpoint.sh]: `.com` for US? I don't think we use `.us` anywhere.

File: lib/test-endpoint.sh (hunk 1 above)
Comment (nmuesch) [reply lib/test-endpoint.sh]: It lets me not explicitly string match in more than one place, which I think is a bit cleaner, with the expense of this variable.

File: lib/test-endpoint.sh [hunk 4]
```diff
@@ -2,13 +2,26 @@
 
 unset DD_LOGS_VALID_ENDPOINT
 DATADOG_DIR="${DATADOG_DIR:-/home/vcap/app/datadog}"
+DD_EU_API_SITE="https://api.datadoghq.eu/api/"
+DD_US_API_SITE="https://api.datadoghq.com/api/"
+DD_API_SITE=${DD_US_API_SITE}
+DD_USE_EU=false
+
+if [ $DD_SITE == "datadog.eu" ]; then
```
Comment (zippolyte) [inline lib/test-endpoint.sh]:
```suggestion
if [ "$DD_SITE" = "datadog.eu" ]; then
```

File: lib/test-endpoint.sh (hunk 3 above)
Comment (dabcoder) [reply lib/test-endpoint.sh]: It's actually `443` in EU indeed: https://docs.datadoghq.com/logs/log_collection/?tab=eusite#datadog-logs-endpoints and `10516` in US: https://docs.datadoghq.com/logs/log_collection/?tab=ussite#datadog-logs-endpoints

File: lib/test-endpoint.sh [hunk 5]
```diff
@@ -2,11 +2,23 @@
...
-if [ -z $DD_LOGS_CONFIG_LOGS_DD_URL ]; then
+if [[ -n $DD_SITE ]] && [[ "$DD_SITE" == "datadoghq.eu" ]]; then
+  DD_USE_EU=true
+  DD_API_SITE=$DD_EU_API_SITE
+fi
+
+if [[ -z "$DD_LOGS_CONFIG_LOGS_DD_URL" ]]; then
   # Initialize to default value
-  if [ -n $DD_LOGS_CONFIG_DD_PORT -a -n $DD_LOGS_CONFIG_DD_URL ]; then
+  # If DD_SITE contains ".eu" use the EU endpoint, otherwise default to US
```
Comment (gzussa) [inline lib/test-endpoint.sh]: This description doesn't seem to match what the source code below does, does he?

Comment (nmuesch) [reply lib/test-endpoint.sh]: 🤦‍♂ Good catch, updated.

**Pull request merged**